qa/tasks/barbican.py

```diff
@@ -157,11 +158,15 @@ def fix_barbican_api(ctx, cclient):
                          '/prop_dir =/ s#etc/barbican#{}/etc/barbican#'.format(get_barbican_dir(ctx)),
                          'bin/barbican-api'])
 
-def copy_policy_json(ctx, cclient):
-    run_in_barbican_dir(ctx, cclient,
-                        ['cp',
-                         get_barbican_dir(ctx)+'/etc/barbican/policy.json',
-                         get_barbican_dir(ctx)])
+def copy_policy_file(ctx, cclient):
+    policy_json_path = get_barbican_dir(ctx)+'/etc/barbican/policy.json'
+    if os.path.exists(policy_json_path):
+        run_in_barbican_dir(ctx, cclient,
+                            ['cp', policy_json_path, get_barbican_dir(ctx)])
```

Comment:
i don't think you'll be able to rely on the local filesystem this way for `os.path.exists()`. this `qa/tasks/barbican.py` script will be running somewhere like `teuthology.front.sepia.ceph.com`, but `run_in_barbican_dir()` is shelling into one of the worker nodes to execute these commands

instead of testing whether the paths exist first, you could run both commands and ignore errors:
```py
        run_in_barbican_dir(ctx, cclient,
                            ['cp', policy_json_path, get_barbican_dir(ctx)],
                            check_status=False) # ignore file not found
```
`run_in_barbican_dir()` doesn't support `check_status` yet, so you'd have to modify it to pass along `**kwargs` like `run_in_keystone_dir()` does in https://github.com/ceph/ceph/blob/main/qa/tasks/keystone.py#L25-L29

Reply:
Ack, I think this is old logic though and we don't even need the file since the default policy is in the code in newer versions. Good thing is that rgw:crypt suite's barbican task is now passing, but I will try removing it and reschedule the job.

http://pulpito.front.sepia.ceph.com/tobias-urdin-2022-09-22_12:49:10-rgw:crypt-main-distro-default-smithi/#

I will also schedule a new tempest suite after the change, if everything passes we should be good here.